storage/client_test.go

```diff
@@ -337,16 +337,21 @@ func TestUpdateRangeAddressing(t *testing.T) {
 	for i, test := range testCases {
 		left := &proto.RangeDescriptor{RaftID: int64(i * 2), StartKey: test.leftStart, EndKey: test.leftEnd}
 		right := &proto.RangeDescriptor{RaftID: int64(i*2 + 1), StartKey: test.rightStart, EndKey: test.rightEnd}
+		var calls []*client.Call
 		if test.split {
-			if err := storage.SplitRangeAddressing(store.DB(), left, right); err != nil {
+			var err error
+			if calls, err = storage.SplitRangeAddressing(left, right); err != nil {
```

Comment:
I take it as a given that we do not want `KV` to contain a list of `prepared` statements as doing so makes `KV` non-thread-safe. Having separate `KV` and `Txn` objects does not bother me. That's the same pattern used by the `database/sql` pattern and it makes sense that there is a somewhat different operation mix between the two. That said, the existence of `Txn.{Prepare,Flush}` is a bit bothersome.

Here's another idea: we add a `client.Batch` struct that holds a slice of calls. Then we could have a structure like:

```
type KV struct { ... }
func (*KV) Run(calls ...*Call) error {}
func (*KV) Exec(b *Batch) error {}

type Txn struct { ... }
func (*Txn) Run(calls ...*Call) error {}
func (*Txn) Exec(b *Batch) error {}

type Batch struct { }
func (*Batch) Run(calls ...*Call) error {}
```

So `KV`, `Txn` and `Batch` would all implement a `Run(calls ...*Call)` method and thus be interchangeable in must situations via an interface.